modelcluster/queryset.py

```diff
@@ -516,6 +517,21 @@ def order_by(self, *fields):
         clone = self.get_clone(results=self.results[:])
         sort_by_fields(clone.results, fields)
         return clone
+    
+    def distinct(self, *fields):
+        if fields and connection.vendor != 'postgresql':
+            raise NotSupportedError("DISTINCT ON fields is not supported by this database backend")
+
+        unique_results = []
+        if not fields:
+            fields = [field.name for field in self.model._meta.fields if not field.primary_key]
+        seen_keys = set()
+        for result in self.results:
+            key = '$$$'.join([str(extract_field_value(result, field)) for field in fields])
```

Comment:
I suggest using a tuple here - unlike lists, they're valid to use as keys within a set:
```suggestion
            key = tuple(str(extract_field_value(result, field)) for field in fields)
```